### PR TITLE
Add tests for `ahr_blinded()`

### PR DESCRIPTION
**Pull request by DMuriuki**
Adding test file for the `ahr_blinded()` function.

### CODE REVIEW

**Comment by nanxstats** [comment]
Focus on `tests/testthat/test-independent-ahr_blinded.R`

**Review by nanxstats — CHANGES_REQUESTED** [review]
Hey @DMuriuki I rebased your PR so the merge conflicts are now resolved. Please take a look at this failing test and make necessary updates:

```
  ── Failure ('test-independent-ahr_blinded.R:17:3'): Piecewise model hr and intervals must be aligned ──
  `ahr_blinded(hr = c(1, 3), intervals = c(3, 4))` did not throw an error.
```

You can use

```
git clone -b ahr-blinded-test https://github.com/Merck/gsDesign2.git
```

to clone the latest version of this branch to local, install, fix the test, and push.

**Review by nanxstats — COMMENTED** [review]
@LittleBeannie This PR is ready for review. 90% code coverage for `ahr_blinded()`.

File: tests/testthat/test-independent-ahr_blinded.R [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    event = simtrial::ex1_delayed_effect$evntd
+  )
+  intervals <- c(3, 6, Inf)
+  hr <- c(1, 0.7, 0.5)
+  ratio <- 2
+
+  # Run the function
+  result <- ahr_blinded(surv = surv, intervals = intervals, hr = hr, ratio = ratio)
+
+  # Test 1: Correct fitting of survival data into piecewise exponential model
```
Comment (LittleBeannie) [inline tests/testthat/test-independent-ahr_blinded.R:31]: I am wondering if "Test 1" is informative or not... it looks like it does not compare the output of `ahr_blinded` with some double-programming/independent programming results...

File: tests/testthat/test-independent-ahr_blinded.R [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  event <- simtrial::fit_pwexp(surv, intervals)[, 3]
+  expect_length(event, length(intervals))
+  expect_true(all(event >= 0))
+
+  # Test 2: Hazard ratio vector is correctly extended
```
Comment (LittleBeannie) [inline tests/testthat/test-independent-ahr_blinded.R:36]: Echo with my comments to "Test 1".

File: tests/testthat/test-independent-ahr_blinded.R [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  nhr <- length(hr)
+  nx <- length(event)
+  if (length(hr) < length(event)) {
+    hr <- c(hr, rep(hr[nhr], nx - nhr))
+  }
+  expect_equal(length(hr), length(event))
+  expect_equal(hr, c(1, 0.7, 0.5)) # Expected extended hr vector
+
+  # Test 3: Blinded AHR (theta) is computed correctly
```
Comment (LittleBeannie) [inline tests/testthat/test-independent-ahr_blinded.R:45]: Echo with my comments to "Test 1".

File: tests/testthat/test-independent-ahr_blinded.R [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  theta <- -sum(log(hr[1:nx]) * event) / sum(event)
```
Comment (LittleBeannie) [inline tests/testthat/test-independent-ahr_blinded.R:46]: It looks like the calculation of `theta` is exactly the same as what we have in https://github.com/Merck/gsDesign2/blob/main/R/ahr_blinded.R#L109. So, I am wondering if it is a validated independent testing or not.

File: tests/testthat/test-independent-ahr_blinded.R [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+  expect_true(!is.na(theta))
+
+  # Test 4: Information adjustment (q_e) is computed correctly
```
Comment (LittleBeannie) [inline tests/testthat/test-independent-ahr_blinded.R:49]: Echo with my comment to "Test 1".

File: tests/testthat/test-independent-ahr_blinded.R [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+  expected_theta <- -sum(log(hr[1]) * sum(surv[, "status"])) / sum(surv[, "status"])
+  expect_equal(result$theta, expected_theta)
+  expect_equal(result$event, sum(surv[, "status"]))
+})
+
+test_that("ahr_blinded handles very small or very large hazard ratios", {
+  surv <- survival::Surv(time = c(1, 2, 3, 4, 5), event = c(1, 1, 1, 1, 1))
+  intervals <- c(2, 3, Inf)
+  hr_small <- c(0.1, 0.2, 0.3)
+  hr_large <- c(10, 20, 30)
```
Comment (LittleBeannie) [inline tests/testthat/test-independent-ahr_blinded.R:124]: HR is always a number between 0 and 1. So it is not statistically meaningful to have `hr_large = c(10, 20, 30)`.

File: tests/testthat/test-independent-ahr_blinded.R [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+  surv <- survival::Surv(time = c(1, 2, 3, 4, 5), event = c(1, 1, 1, 1, 1))
+  intervals <- c(2, 3, Inf)
+  hr <- c(0.8, 0.9, 0.95)
+  ratio <- 100
+
+  result <- ahr_blinded(surv = surv, intervals = intervals, hr = hr, ratio = ratio)
+
+  q_e <- ratio / (1 + ratio)
+  # q_e should be close to 1
+  expect_equal(q_e, 1, tolerance = 0.01)
```
Comment (LittleBeannie) [inline tests/testthat/test-independent-ahr_blinded.R:146]: Echo with my comments to "Test 1".

**Review by LittleBeannie — CHANGES_REQUESTED** [review]
Thanks, @DMuriuki and @nanxstats for the testing! Following, please find my comments.

**Comment by nanxstats** [comment]
@LittleBeannie Great, I've just removed the non-informative and unrealistic test cases.

**Review by LittleBeannie — APPROVED** [review]
Thanks a lot, @DMuriuki and @nanxstats !

**Pull request merged**